pkg/controllers/binding/binding_controller.go

```diff
@@ -111,6 +105,11 @@ func (c *ResourceBindingController) syncBinding(binding *workv1alpha2.ResourceBi
 		return controllerruntime.Result{Requeue: true}, err
 	}
 
+	// no need to dispatch works if no clusters scheduled.
+	if len(clusterNames) == 0 {
```

Comment:
Well, I don't think it's a good idea to use `len(clusterNames)==0` to check if the binding has been **scheduled once**, though it works for now.
A better solution is https://github.com/karmada-io/karmada/pull/988#issuecomment-982445685

Reply:
> 

We don't care about if the binding has been scheduled once,now we will always sync binding. we reture if `len(clusterNames) == 0` because when `len(clusterNames) == 0`, all reference works are orphaned and will be deleted. So, we don't need to  aggregate status. Meanwhile, if `len(clusterNames) == 0` means no cluster matched which won't create new works.

Reply:
> all reference works are orphaned and will be deleted. So, we don't need to aggregate status

Why? As I said above:
> If there is no cluster that fits an **updated** propagation policy, `len(targetClusters)` would be 0, then binding controller would not be able to aggregate status

If we don't aggregate status, the binding status would stay stale, which seems incorrect

I know this cannot happen for now, but [here](https://github.com/karmada-io/karmada/pull/988#issuecomment-982431007) is my concern

Reply:
> 

Got it! you mean when `len(clusterNames) == 0 `, we should clean the old status! It seems like the PolicyPlacement Annotation is the only way to indicate a cluster has been scheduled once?

Reply:
Exactly!

Reply:
> 

Seems like two ways to solve this problem:
1. use the PolicyPlacement Annotation to check if the binding has been scheduled once. @RainbowMango think it is weird
2. always sync binding no matter whether the binding has been scheduled once or not

Reply:
I vote for option `2`.

Reply:
Either is ok for me